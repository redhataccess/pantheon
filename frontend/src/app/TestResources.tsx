import { IAppState } from './app';

const fakeState: IAppState = {
    isNavOpen: true,
    userAuthenticated: true,
    username: "demo"
}

export { fakeState }