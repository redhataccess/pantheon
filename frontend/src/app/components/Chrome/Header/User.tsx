import React, { Component } from 'react';
import { Link } from "react-router-dom";

class User extends Component<any, any> {
    constructor(props) {
        super(props);
        this.state = {
            isLoggedIn: false,
            linkText: 'Log In'
        };
    }

    public componentDidMount() {

        if (!this.state.isLoggedIn) {
            fetch("/system/sling/info.sessionInfo.json")
                .then(response => response.json())
                .then(responseJSON => {
                    if (responseJSON.userID !== 'anonymous') {
                        this.setState({ linkText: 'Log Out [' + responseJSON.userID + ']' })
                        this.setState({ isLoggedIn: true })
                    }
                })
        }
    }

    public render() {
        return (
            <React.Fragment>
                <Link to={this.state.isLoggedIn ? '' : '/login'}
                    onClick={this.conditionalRedirect}>
                    {this.state.linkText}
                </Link>
            </React.Fragment>
        );
    }

    private conditionalRedirect = () => {
        if (this.state.linkText.includes("Log Out")) {
            fetch('/system/sling/logout')
                .then(response => window.location.href = "/pantheon")
        }
    }
}

export { User }