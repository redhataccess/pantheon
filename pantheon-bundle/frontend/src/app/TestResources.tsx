import { IAppState } from './app';

const mockStateUser: IAppState = {
    isAdmin: false,
    isNavOpen: true,
    userAuthenticated: true,
    username: "demo"
}

const mockStateAdmin: IAppState = {
    isAdmin: true,
    isNavOpen: true,
    userAuthenticated: true,
    username: "admin"
}

const mockStateGuest: IAppState = {
    isAdmin: false,
    isNavOpen: true,
    userAuthenticated: false,
    username: "anonymous"
}

export { mockStateUser, mockStateAdmin, mockStateGuest }