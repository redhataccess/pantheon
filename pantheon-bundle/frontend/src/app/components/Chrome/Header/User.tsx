import React, { Component } from "react"
import { Link } from "react-router-dom"
import {
    Button, Dropdown,
    DropdownToggle,
    DropdownItem, Modal, DropdownPosition
} from "@patternfly/react-core"
import { HelpIcon } from "@patternfly/react-icons"
import "@app/app.css"
import { IAppState } from "@app/app"

interface IState {
    helpDropdownOpen: boolean
    loginUrl: string
    logoutUrl: string
}

class User extends Component<IAppState, IState> {
    constructor(props) {
        super(props)
        this.state = {
            helpDropdownOpen: false,
            loginUrl: "/auth/login",
            logoutUrl: "/system/sling/logout"
        }
    }

    public render() {
        const dropdownItems = [
            <DropdownItem key="userguide" href="/pantheon/docs/assemblies/assembly-pantheon-help.html" target="_blank">User Guide</DropdownItem>,
            <DropdownItem key="contribute" href="https://github.com/redhataccess/pantheon" target="_blank">Contribute to Pantheon</DropdownItem>
        ]
        return (
            <React.Fragment>
                <Dropdown onSelect={this.onHelpSelect}
                    toggle={
                        <DropdownToggle toggleIndicator={null} onToggle={this.onHelpToggle}>
                            <HelpIcon />
                        </DropdownToggle>
                    }
                    isPlain={true}
                    isOpen={this.state.helpDropdownOpen}
                    dropdownItems={dropdownItems}
                    position={DropdownPosition.right}
                />
                <a className="p2-header__login"
                    href={this.props.userAuthenticated ? this.state.logoutUrl : this.state.loginUrl}
                    onClick={this.conditionalRedirect}>
                    {this.props.userAuthenticated ? "[" + this.props.username + "]" : "Log In"}
                </a>
            </React.Fragment>
        )
    }

    private onHelpToggle = helpDropdownOpen => {
        this.setState({ helpDropdownOpen })
    }

    private onHelpSelect = () => {
        this.setState({
            helpDropdownOpen: !this.state.helpDropdownOpen
        })
    }

    private conditionalRedirect = () => {
        if (this.props.userAuthenticated) {
            fetch(this.state.logoutUrl)
                .then(response => window.location.href = "/pantheon")
        }
    }
}

export { User }
