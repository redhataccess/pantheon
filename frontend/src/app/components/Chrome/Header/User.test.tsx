import React from 'react';
import { User } from './User';
import "isomorphic-fetch"
import { Link } from "react-router-dom";
import { HashRouter as Router } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { mockStateUser } from '@app/TestResources'

describe('User tests', () => {
  test('should render User component', () => {
    const view = shallow(<User {...mockStateUser} />);
    expect(view).toMatchSnapshot();
  });

  it('should render a Link component', () => {
    const wrapper = mount(<Router><User {...mockStateUser} /></Router>);
    const navLinks = wrapper.find(Link);
    expect(navLinks.exists()).toBe(true)
  });

  it('test render function', () => {
    const wrapper = renderer.create(<Router><User {...mockStateUser} /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.render).toMatchSnapshot();
  });

  it('test conditionalRedirect function', () => {
    const wrapper = renderer.create(<Router><User {...mockStateUser} /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.conditionalRedirect).toMatchSnapshot();
  });

  it('test componentDidMount function', () => {
    const wrapper = renderer.create(<Router><User {...mockStateUser} /></Router>);
    const inst = wrapper.getInstance();
    expect(inst.componentDidMount).toMatchSnapshot();
  });
});
