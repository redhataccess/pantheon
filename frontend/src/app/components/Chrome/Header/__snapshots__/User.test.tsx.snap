// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User tests should render User component 1`] = `
<Fragment>
  <Link
    onClick={[Function]}
    to="/login"
  >
    Log In
  </Link>
</Fragment>
`;

exports[`User tests test componentDidMount function 1`] = `[Function]`;

exports[`User tests test conditionalRedirect function 1`] = `undefined`;

exports[`User tests test render function 1`] = `[Function]`;
